pkg/etw/provider.go

```diff
@@ -120,7 +120,13 @@ func providerIDFromName(name string) guid.GUID {
 // NewProvider creates and registers a new ETW provider. The provider ID is
 // generated based on the provider name.
 func NewProvider(name string, callback EnableCallback) (provider *Provider, err error) {
-	return NewProviderWithID(name, providerIDFromName(name), callback)
+	return NewProviderWithOptions(name, WithCallback(callback))
+}
+
+// NewProviderWithGroup creates and registers a new ETW provider with a
+// specificed group. The provider ID is generated based on the provider name.
+func NewProviderWithGroup(name string, group guid.GUID, callback EnableCallback) (provider *Provider, err error) {
```

Comment:
I think it's fine to not add this. We can consider `NewProvider` and `NewProviderWithID` as legacy, and treat `NewProviderWithOptions` as the single entry-point for creating a provider going forward.